frontend/src/pages/clusterSettings/ModelServingPlatformSettings.tsx

```diff
@@ -155,12 +161,12 @@ const ModelServingPlatformSettings: React.FC<ModelServingPlatformSettingsProps>
                       {
                         key: DeploymentMode.RawDeployment,
                         label: 'Standard (No additional dependencies)',
-                        isDisabled: true, // todo: allow admin to update dsc
+                        isDisabled: !allowedToPatchDSC,
                       },
                       {
                         key: DeploymentMode.Serverless,
                         label: 'Advanced (Serverless and Service Mesh)',
-                        isDisabled: !isServerlessAvailable || true, // todo: allow admin to update dsc
+                        isDisabled: !isServerlessAvailable && !allowedToPatchDSC,
```

Comment:
```suggestion
                        isDisabled: !isServerlessAvailable || !allowedToPatchDSC,
```
This should be an `||` because if serverless is not available, you shouldn't be able to select it